src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -597,9 +597,9 @@ public static CSharpType FromLiteral(CSharpType type, object literalValue)
         /// <param name="unionItemTypes">The list of union item types.</param>
         /// <param name="isNullable">Flag used to determine if a type is nullable.</param>
         /// <returns>A <see cref="CSharpType"/> instance representing those unioned types.</returns>
-        public static CSharpType FromUnion(IReadOnlyList<CSharpType> unionItemTypes, bool isNullable)
+        public static CSharpType FromUnion(IReadOnlyList<CSharpType> unionItemTypes)
```

Comment:
I think this is irrelavent to your changes.
Like we never remove the `IsNullable` in CSharpType, we should remove it here.
But it makes more sense if it is optional here

Reply:
I changed it to optional parameter